### PR TITLE
[oneDNN] Fixing eager-related nightly test failures

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes : //tensorflow/python/kernel_tests:matmul_op_test and run_eager_op_as_function_test,
These tests which are run with eager op as function enabled, fail with error , "Could not find device for node:"

With a recent commit on public tensorflow : tensorflow/tensorflow@1224326
Eager ops are wrapped, inside function WrapInCallOp, after the device has been inferred for op.

For MKL : we expect WrapInCallOp to be called before we infer device, which marks the '_kernel' attribute as NameChangeOp label for nodeDef of funcDef.
Since with new commit, WrapInCallOp is called AFTER the device inference, no device is found for the MKL Op as no kernel match is found without namechange label.

In this PR: we mark nodeDef with '_kernel' attribute as NameChange label at both places : a) before inferring the device and b) inside WrapInCallOp (already present in current code base)

### CODE REVIEW

**Comment by kkimdev** [comment]
@saxenasaurabh It sounds like related to the eager-op-as-function project, Could you take a look at this?

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 1]
```diff
@@ -727,6 +727,16 @@ Status PopulateRetMap(FunctionDef* fdef, const AbstractOpAttrs* op_attrs,
   return Status::OK();
 }
 
+#ifdef INTEL_MKL
+void GetMKLNodeDef(NodeDef* ndef) {
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:731]: Can we make this an inline function?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the fix!

**Comment by sachinmuradi** [comment]
@googlebot I consent.

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 1 above)
Comment (sachinmuradi) [reply tensorflow/core/common_runtime/eager/execute.cc]: Changes done.

**Review by penpornk — APPROVED** [review]
Thank you for the change!

**Pull request merged**